include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,465 @@
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary %= operator to compute/assign remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator %=(Angle& left, Angle right);
+
+inline namespace literals
+{
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees, e.g. 10.5_deg
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees, e.g. 90_deg
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(unsigned long long angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians, e.g. 0.1_rad
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians, e.g. 2_rad
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(unsigned long long angle);
+
+} // namespace literals
+
+#include <SFML/System/Angle.inl>
+
+} // namespace sf
+
+
+#endif // SFML_ANGLE_HPP
+
+
+////////////////////////////////////////////////////////////
+/// \class sf::Angle
+/// \ingroup system
+///
+/// sf::Angle encapsulates an angle value in a flexible way.
+/// It allows to define an angle value either as a number of
+/// degrees or radians. It also works the other way round:
+/// you can read an angle value as either a number of
+/// degrees or radians.
+///
+/// By using such a flexible interface, the API doesn't
+/// impose any fixed type or unit for angle values, and let
+/// the user choose their own favorite representation.
+///
+/// Angle values support the usual mathematical operations:
+/// you can add or subtract two angles, multiply or divide
+/// an angle by a number, compare two angles, etc.
+///
+/// Usage example:
+/// \code
+/// sf::Angle a1  = sf::degrees(90.0f);
+/// float radians = a1.asRadians(); // 1.5708f
+///
+/// sf::Angle a2 = sf::radians(3.141592654f);
+/// float degrees = a2.asDegree(); // 180.0f
+/// \endcode
```

Comment:
IMO, it would be nice to have an example of using user defined literals here, showing `using namespace sf::literals`.

Reply:
Good idea. Do you think this is sufficient?
```cpp
/// using namespace sf::literals;
/// sf::Angle a3 = 10_deg; // 10 degrees
/// sf::Angle a4 = 1_rad;  // 1 radian
```

Reply:
Nice. Maybe have `1.42_rad` or something to illustrate both integer and floating literals.

Reply:
Done. Now it's this:
```cpp
/// using namespace sf::Literals;
/// sf::Angle a3 = 10_deg;   // 10 degrees
/// sf::Angle a4 = 1.5_deg;  // 1.5 degrees
/// sf::Angle a5 = 1_rad;    // 1 radian
/// sf::Angle a6 = 3.14_rad; // 3.14 radians
```